src/deploy/firestore/release.ts

```diff
@@ -1,25 +1,36 @@
 import * as _ from "lodash";
 
 import { RulesDeploy, RulesetServiceType } from "../../rulesDeploy";
+import { logger } from "../../logger";
 import { Options } from "../../options";
-import { FirebaseError } from "../../error";
 
 /**
  * Releases Firestore rules.
  * @param context The deploy context.
  * @param options The CLI options object.
  */
-export default async function (context: any, options: Options): Promise<void> {
+export default async function (context: any, options: Options): Promise<any> {
   const rulesDeploy: RulesDeploy = _.get(context, "firestore.rulesDeploy");
   if (!context.firestoreRules || !rulesDeploy) {
-    return;
+    return Promise.resolve();
   }
 
-  const rulesFile = options.config.src.firestore?.rules;
-  if (!rulesFile) {
-    throw new FirebaseError(
-      `Invalid firestore config: ${JSON.stringify(options.config.src.firestore)}`
-    );
-  }
-  await rulesDeploy.release(rulesFile, RulesetServiceType.CLOUD_FIRESTORE);
+  const rulesContext = _.get(context, "firestore.rules");
+  return Promise.all(
+    rulesContext.map((ruleContext: any) => {
```

Comment:
Suggest making this an async function - then, you can omit the Promise.resolves inside it and just return what you want